### PR TITLE
Testing improvements

### PR DESCRIPTION
**Pull request by WillB97**
- Adds CI, runnable manually and on tags
- Re-enables DHCP with a fallback IP on eth0
- Adds robot user to adm group so that it can view journalctl
- Keeps build-essential, this is useful when using pip

There will be other changes that need making when sbot gets updated. Equally we should probably explore replacing usbmount with udiskie which can be installed with apt.

### CODE REVIEW

File: parts/cleanup.sh [hunk 1]
```diff
@@ -1,10 +1,9 @@
 #!/bin/bash
 set -eux -o pipefail
 
-systemctl disable dhcpcd
 systemctl disable hciuart
 
-apt-get remove --yes build-essential devscripts
```
Comment (RealOrangeOne) [inline parts/cleanup.sh:7]: What's the reason for keeping `build-essential` on the device? There shouldn't be much to build once the image has built?

File: .github/workflows/build-docker.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+      - name: Checkout
+        uses: actions/checkout@v3.3.0
+        with:
+          fetch-depth: 0
+          submodules: recursive
+
+      - name: Build image
+        run: |
+          chmod +x build-image.sh
```
Comment (RealOrangeOne) [inline .github/workflows/build-docker.yml]: Nit: Script is already executable

```suggestion
```

File: parts/base.sh [hunk 3]
```diff
@@ -23,9 +23,12 @@ cat >>/boot/config.txt <<EOF
 enable_uart=1
 EOF
 
-cat >>/etc/network/interfaces <<EOF
-# Bring up interface with link local address
-# As DHCPCD is disabled, we can disregard the above warning
-allow-hotplug eth0
-iface eth0 inet manual
+cat >>/etc/dhcpcd.conf <<EOF
+# define a static profile to use of DHCP fails
```
Comment (RealOrangeOne) [inline parts/base.sh]: Nit: Typo
```suggestion
# define a static profile to use if DHCP fails
```

File: parts/cleanup.sh (hunk 1 above)
Comment (WillB97) [reply parts/cleanup.sh:7]: When installing things from pip that have c-bindings you tend to need this or you get strange errors while installing. Generally it's useful and not massive.

**Comment by WillB97** [comment]
We should also compress the image otherwise it takes a really long time to download the image while github zips it.

File: parts/cleanup.sh (hunk 1 above)
Comment (RealOrangeOne) [reply parts/cleanup.sh:7]: It's a meta package with a lot in it. We shouldn't need to be installing things after build, right?

Comment (WillB97) [reply parts/cleanup.sh:7]: It's only 45MB and gives us flexibility,

**Pull request merged**